modules/matterhorn-workflow-service-impl/src/main/java/org/opencastproject/workflow/impl/WorkflowServiceImpl.java

```diff
@@ -2392,15 +2391,11 @@ public synchronized void cleanupWorkflowInstances(int buffer, WorkflowState stat
 
   @Override
   public void repopulate(final String indexName) throws Exception {
-    List<Job> jobs = null;
+    List<Job> jobs = new ArrayList<>();
     try {
-      jobs = serviceRegistry.getJobs(WorkflowService.JOB_TYPE, null);
-      Iterator<Job> ji = new ArrayList<>(jobs).iterator();
-      while (ji.hasNext()) {
-        Job job = ji.next();
-        if (!WorkflowServiceImpl.Operation.START_WORKFLOW.toString().equals(job.getOperation())) {
-          logger.debug("Removing unrelated job {} of type {}", job.getId(), job.getOperation());
-          ji.remove();
+      for (Job job : serviceRegistry.getJobs(WorkflowService.JOB_TYPE, null)) {
```

Comment:
Hmmm.... would be very desirable to not retrieve all jobs (IIRC, that can be millions) just the ignore most of them here.... but that's not in scope of this PR. I've filed a JIRA issue for this: https://opencast.jira.com/browse/MH-12640